pkg/pillar/cmd/volumemgr/handledownloader.go

```diff
@@ -25,17 +28,37 @@ func AddOrRefcountDownloaderConfig(ctx *volumemgrContext, status types.VolumeSta
 	} else {
 		log.Debugf("AddOrRefcountDownloaderConfig: add for %s\n",
 			status.BlobSha256)
+
+		name := status.DownloadOrigin.Name
+
+		// where should the final downloaded file be?
+		locFilename := path.Join(types.DownloadDirname, status.ObjType, "pending", status.VolumeID.String(), path.Base(name))
```

Comment:
We don't yet have a volumeid for the volumes.
Downloader code uses status.ImageID.String() to make the not-quite unique name.
ote that we now know the claimed sha (from the controller or from resolving the tag) hence we can use that to get a unique pathname combined with the imageID.

Reply:
Note that the code below using imageid and not volumeid to determine the filename.

Reply:
That is a very good point. We are downloading a unique _image_, so its download should be tied to the _imageID_ not to the _volumeID_, which could be one of several that will be generated from the image. Fixing.

Reply:
No, now I am confused. The original code here from this func is [unchanged](https://github.com/lf-edge/eve/pull/958/files#diff-f461cd497d76c483a20f5054d84021ccR46-R48):

```go
		n := types.DownloaderConfig{
			ImageID:     status.VolumeID,
			DatastoreID: status.DownloadOrigin.DatastoreID,
```

So `config.ImageID` is derived from `status.VolumeID`, and the old code that generated the directory is [here](https://github.com/lf-edge/eve/pull/958/files#diff-b4ed0369ba5dc47d9f41246487b8efb1L48-L51):

```go
	locDirname := types.DownloadDirname + "/" + status.ObjType
 	locFilename = locDirname + "/pending"
 	// XXX common routines to determine pathnames?
 	locFilename = locFilename + "/" + config.ImageID.String()
```

So it creates the dir from `config.ImageID`, but that was set from `status.VolumeID`. I just carried it up earlier. What am I missing?

Reply:
Turns out that field is only used for logging as it says
	ImageID          uuid.UUID // Used for logging

So it is a bug to use VolumeID, but a rather harmless one.

Reply:
> Turns out that field is only used for logging as it says

I don't see that. The code in downloader used `ImageID` to make the path for download, does it not?

Reply:
They shouldn't be used based on
types/downloadertypes.go:	ImageID          uuid.UUID // Used for logging
types/downloadertypes.go:	ImageID          uuid.UUID // Used for logging
types/verifiertypes.go:	ImageID     uuid.UUID // Used for logging
types/verifiertypes.go:	ImageID       uuid.UUID // Used for logging

Reply:
I am not disagreeing that they shouldn't be used based on the comments. But it looks like it _is_ used, and that is how it sets up the file path for downloads (eventually getting moved to `verifier/` and `verified/`.

In truth, it does not really matter what we use, as downloader becomes "dumb" about it, and only volumemgr controls it. I will see if we can extract and put the actual image ID in there.